src/shared/ui/components/widget-tab/widget-tab.component.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useMemo } from 'react';
+
+import { useWidgetTabs } from '../../providers/widget-tabs';
+import { WidgetBase } from '../widget-base';
+import { classes } from '../../utils/classes';
+
+import { TabHandle } from './tab-handle.component';
+import { WidgetTabProperties } from './widget-tab.types';
+
+export const WidgetTab = ({
+  top,
+  theme,
+  twitterHandle,
+  tabImage,
+  tabName,
+  children,
+  className,
+  closeButtonClassName,
+  onClose,
+}: WidgetTabProperties) => {
+  const {
+    tabs,
+    preferredTab,
+    addWidgetTab,
+    setUserPreferredTab,
+    removeAllUserWidgets,
+  } = useWidgetTabs();
+
+  const userTabs = useMemo(() => {
+    return tabs[twitterHandle];
+  }, [tabs, twitterHandle]);
+
+  const userPreferredTab = preferredTab[twitterHandle] ?? userTabs?.at(0);
+
+  const left = useMemo(() => {
+    const index = userTabs?.indexOf(tabName) ?? -1;
+    if (index !== -1) {
+      return index * 100;
+    }
+
+    return 0;
+  }, [tabName, userTabs]);
+
+  useEffect(() => {
+    addWidgetTab(twitterHandle, tabName);
+  }, [addWidgetTab, tabName, twitterHandle]);
+
+  useEffect(() => {
+    if (userTabs === null && onClose) {
+      onClose();
+    }
+  }, [onClose, userTabs]);
+
+  if (!userTabs || userTabs.length === 0) {
+    return;
+  }
+
+  return (
+    <WidgetBase
+      className={classes(
+        'z-10 overflow-visible rounded-[0.375rem] text-xs leading-tight ',
+        className,
+        {
+          'z-20': userPreferredTab === tabName,
+          'rounded-tl-none': userTabs.length > 1,
+          'bg-[#2d2d2d]': theme === 'dark',
+          'bg-white': theme === 'bright',
+        },
+      )}
```

Comment:
I am wondering if className shouldn't be the last one because it's something passed from higher layer and higher layer may always want to override some styles, in tailwind and twMerge util (which we use under classess abstraction) latest class will be applied, so imagine you want in some specific case override background to `bg-red` it still won't be applied because later `bg-white` is applied, should we move className as last arg?

Reply:
Agree! Good point